src/main/java/com/hivemq/mqtt/handler/InboundInterceptorHandler.java

```diff
@@ -0,0 +1,79 @@
+package com.hivemq.mqtt.handler;
+
+import com.hivemq.extension.sdk.api.annotations.NotNull;
+import com.hivemq.extensions.handler.*;
+import com.hivemq.mqtt.message.PINGREQ;
+import com.hivemq.mqtt.message.connect.CONNECT;
+import com.hivemq.mqtt.message.disconnect.DISCONNECT;
+import com.hivemq.mqtt.message.puback.PUBACK;
+import com.hivemq.mqtt.message.pubcomp.PUBCOMP;
+import com.hivemq.mqtt.message.pubrec.PUBREC;
+import com.hivemq.mqtt.message.pubrel.PUBREL;
+import com.hivemq.mqtt.message.unsubscribe.UNSUBSCRIBE;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * @author Daniel Krüger
+ */
+@Singleton
+@ChannelHandler.Sharable
+public class InboundInterceptorHandler extends ChannelInboundHandlerAdapter {
+
+    private final @NotNull ConnectInboundInterceptorHandler connectInboundInterceptorHandler;
+    private final @NotNull UnsubscribeInboundInterceptorHandler unsubscribeInboundInterceptorHandler;
+
+    private final @NotNull PingInterceptorHandler pingInterceptorHandler;
+    private final @NotNull PubackInterceptorHandler pubackInterceptorHandler;
+    private final @NotNull PubrecInterceptorHandler pubrecInterceptorHandler;
+    private final @NotNull PubcompInterceptorHandler pubcompInterceptorHandler;
+    private final @NotNull PubrelInterceptorHandler pubrelInterceptorHandler;
+
+    private final @NotNull DisconnectInterceptorHandler disconnectInterceptorHandler;
```

Comment:
I really like ordering by the MQTT message type (CONNECT, CONNACK, PUBLISH, PUBACK, PUBREC, PUBREL, PUBCOMP, SUBSCRIBE, SUBACK, UNSUBSCRIBE, UNSUBACK, PINGREQ, PINGRESP, DISCONNECT, AUTH)

Reply:
the same order should then also be used for the constructor arguments

Reply:
yeah, makes sense.